src/test/java/com/acrolinx/sidebar/utils/MatchUtilsTest.java

```diff
@@ -25,4 +27,63 @@ void testSortByOffsetDesc() {
     assertEquals("2", sortedMatches.get(2).getContent());
     assertEquals("1", sortedMatches.get(3).getContent());
   }
+
+  @Test
+  void shouldRemoveWhiteSpaceMatch() {
+    final List<AcrolinxMatch> acrolinxMatches =
+        List.of(
+            new AcrolinxMatch(new IntRange(0, 3), "abc"),
+            new AcrolinxMatch(new IntRange(3, 6), " "),
+            new AcrolinxMatch(new IntRange(6, 9), "pqr"));
+    String lastCheckedContent = "abc\n\t\tpqr";
+
+    verifyMatches(
+        List.of(acrolinxMatches.get(0), acrolinxMatches.get(2)),
+        MatchUtils.filterInsignificantWhitespaceMatches(acrolinxMatches, lastCheckedContent));
+  }
+
+  @Test
+  void shouldNotRemoveWhiteSpaceMatch() {
+    final List<AcrolinxMatch> acrolinxMatches =
+        List.of(
+            new AcrolinxMatch(new IntRange(0, 3), "abc"),
+            new AcrolinxMatch(new IntRange(3, 4), " "),
+            new AcrolinxMatch(new IntRange(4, 7), "pqr"));
+    String lastCheckedContent = "abc pqr";
+
+    verifyMatches(
+        acrolinxMatches,
+        MatchUtils.filterInsignificantWhitespaceMatches(acrolinxMatches, lastCheckedContent));
+  }
+
+  @Test
+  void shouldNotRemoveWhiteSpacesIfRepresentedCorrectlyByOffsets() {
+    final List<AcrolinxMatch> acrolinxMatches =
+        List.of(
+            new AcrolinxMatch(new IntRange(0, 3), "abc"),
+            new AcrolinxMatch(new IntRange(3, 7), "    "),
+            new AcrolinxMatch(new IntRange(7, 10), "pqr"));
+    String lastCheckedContent = "abc    pqr";
+
+    verifyMatches(
+        acrolinxMatches,
+        MatchUtils.filterInsignificantWhitespaceMatches(acrolinxMatches, lastCheckedContent));
+  }
+
+  private void verifyMatches(
+      List<? extends AbstractMatch> expected, List<? extends AbstractMatch> actual) {
+    assertEquals(expected.size(), actual.size());
+    for (int i = 0; i < expected.size(); i++) {
+      final AbstractMatch expectedMatch = expected.get(i);
```

Comment:
to shorten this method, we can extract the body of the for loop into a separate helper method to just compare two AbstractMatch instances

Reply:
Fixed